be/test/util/uuid_generator_test.cpp

```diff
@@ -0,0 +1,92 @@
+// Copyright 2021-present StarRocks, Inc. All rights reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#include "util/uuid_generator.h"
+
+#include <gtest/gtest.h>
+
+#include <algorithm>
+#include <boost/uuid/uuid_io.hpp>
+#include <cstring>
+#include <thread>
+#include <unordered_set>
+#include <vector>
+
+namespace starrocks {
+class ThreadLocalUUIDGeneratorTest : public testing::Test {
+public:
+    ThreadLocalUUIDGeneratorTest() = default;
+    ~ThreadLocalUUIDGeneratorTest() override = default;
+};
+
+// Test that UUIDs are unique
+TEST_F(ThreadLocalUUIDGeneratorTest, TestUniqueness) {
+    const int NUM_UUIDS = 10000;
+    std::unordered_set<std::string> uuids;
+
+    for (int i = 0; i < NUM_UUIDS; i++) {
+        boost::uuids::uuid uuid = ThreadLocalUUIDGenerator::next_uuid();
+        std::string uuid_string = boost::uuids::to_string(uuid);
+        // Make sure we haven't seen this UUID before
+        ASSERT_TRUE(uuids.find(uuid_string) == uuids.end()) << "Generated duplicate UUID: " << uuid_string;
+
+        uuids.insert(uuid_string);
+    }
```

Comment:
```suggestion
    for (int i = 0; i < NUM_UUIDS; i++) {
        boost::uuids::uuid uuid = ThreadLocalUUIDGenerator::next_uuid();
        std::string uuid_string = boost::uuids::to_string(uuid);
        uuids.insert(uuid_string);
    }
     // Make sure we haven't seen this UUID before
    ASSERT_EQ(NUM_UUIDS, uuids.size()) << "Duplicate UUID generated";
```
This will be more efficient